### DISCUSSION TITLE
Is it possible to add more depots when writing instances in VRPLIB package?

### DISCUSSION THREAD

**Discussion by KatarinaDajic** [PyVRP/PyVRP#575] [Q&A]
Hi guys, I am also getting familiar with the VRPLIB package you use and I am wondering is it possible to add there more depots when creating instances? Also, is the package able to calculate the distance matrix in case of multiple depots?

Thanks for your help!

**Comment by leonlan** [comment]
Hi @KatarinaDajic! 👋 

Yes, it is possible to add multiple depots when creating instances. We have VRPLIB instances for the multi-depot VRP with time windows, which can be found [here](https://github.com/PyVRP/Instances/tree/main/MDVRPTW).

Here's a short example of the PR11A instance:

```
NAME: PR11A
COMMENT: After Vidal et al. (2013) and adapted to PyVRP's VRPLIB dialect.
TYPE: MDVRPTW
EDGE_WEIGHT_TYPE: EUC_2D
DIMENSION: 364
VEHICLES: 40
CAPACITY: 200
VEHICLES_MAX_DURATION: 450
NODE_COORD_SECTION
1	2.958	4.357
2	21.494	-8.63
3	-43.073	-18.094
4	-10.406	-37.888
5	70.769	-29.196
...
DEMAND_SECTION
1	0
2	0
3	0
4	0
5	25
...
SERVICE_TIME_SECTION
1	0
2	0
3	0
4	0
5	20
...
TIME_WINDOW_SECTION
1	0	1000
2	0	1000
3	0	1000
4	0	1000
5	146	281
...
VEHICLES_DEPOT_SECTION
1	1
...
40	5
DEPOT_SECTION
1
2
3
4
EOF
```

- There are four depots, and the location indices corresponding to these depots are specified in the `DEPOT_SECTION`. For all location-related data, the values at indices 1 to 4 correspond to depot data. The client data starts from index 5.
- The `VEHICLES` specification is used to define the number of vehicles, which are 40 vehicles in this case. The home depot of each of these vehicles is given in `VEHICLES_DEPOT_SECTION`.

> Also, is the package able to calculate the distance matrix in case of multiple depots?

I'm not sure whether you are using the `vrplib` package or the `pyvrp.read` function, but both ways support distance matrices with multiple depots.
